src/Login.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react'
+import { useState, ChangeEvent, FormEvent } from 'react';
+import {
+  Text,
+  Button,
+  TextInput,
+  PasswordInput,
+} from '@mantine/core'
+
+interface FormValues {
+  password: string;
+  email: string;
+  confirmPassword?: string;
+}
+
+export default function Login() {
+  const [isRegistered, setIsRegistered] = useState(true)
+  const [formValues, setFormValues] = useState<FormValues>({email: '', password: '', confirmPassword: ''});
+  
+  const handleChangeIsRegistered = (): void => {
+    setIsRegistered(!isRegistered)
+  }
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const {name, value} = event.target;
+    setFormValues((prev) => ({...prev, [name]: value}));
+  }
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    console.log('email: ', formValues.email);
+    console.log('password: ', formValues.password);
+  }
+
+  return (
+    <div>
+      {isRegistered === true ? 
+      <>
+        <Text>Log In</Text>
+        <form onSubmit={handleSubmit}>
```

Comment:
try using something like react-hook-form or formik, it'll make validation a lot easier

Reply:
Good choice! Now there is a one key difference between how formik and react-hook-form checks state, did you run into that when you were looking into it?

Reply:
Honestly not really, both seemed to work pretty well out of the box, and so didn't look too deeply into how they managed state, as both spit out the correct string values in console when testing, the choice came down to which was easier to style with tbh. In looking into how they manage state, it looks like formik keeps track of state in it's formik component and so state is changed that way, and therefore everything inside the formik component re-renders whenever that state changes, where RHF tracks changes in the DOM el in a ref attribute so that state change only causes re-renders to happen in the local ele instead of the entire form. Is this correct?